pkg/envoy/registry/services_test.go

```diff
@@ -63,13 +70,55 @@ var _ = Describe("Test Proxy-Service mapping", func() {
 
 			expectedSvc2 := service.MeshService{
 				Namespace: tests.Namespace,
-				Name:      svcName2,
+				Name:      fmt.Sprintf("%s.%s", podName, svcName2),
 				Port:      tests.ServicePort,
 				Protocol:  "http",
 			}
+
+			// Subdomain gets called in the ListProxyServices
+			expectedSvc1.Subdomain()
+			expectedSvc2.Subdomain()
 			expectedList := []service.MeshService{expectedSvc1, expectedSvc2}
 
-			mockKubeController.EXPECT().GetEndpoints(gomock.Any()).Return(nil, nil).Times(2)
+			mockKubeController.EXPECT().GetEndpoints(gomock.Any()).Return(&v1.Endpoints{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: tests.Namespace,
+					Name:      expectedSvc1.Name,
+				},
+				Subsets: []v1.EndpointSubset{
+					{
+						Addresses: []v1.EndpointAddress{
+							{
+								IP: "8.8.8.8",
+							},
+							{
+								IP: "8.8.8.9",
+							},
+						},
+					},
+				},
+			}, nil).Times(1)
+
+			mockKubeController.EXPECT().GetEndpoints(gomock.Any()).Return(&v1.Endpoints{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: tests.Namespace,
+					Name:      expectedSvc2.Name,
+				},
+				Subsets: []v1.EndpointSubset{
+					{
+						Addresses: []v1.EndpointAddress{
+							{
+								IP:       "8.8.8.9",
```

Comment:
If the IPs here aren't relevant to the test, I'd remove them.

Reply:
The IPs are also used down the callstack in ServiceToMeshServices